File: frontend/src/api/analysisViz.ts
/**
 * 多维可视化：GET /analysis/fund、GET /analysis/trip
 */
import http from './client'

export interface FundTimelineEvent {
  ts: string
  kind: string
  label: string
  amount?: number | null
  from_party?: string | null
  to_party?: string | null
  meta?: Record<string, unknown>
}

export interface FundGraphNode {
  id: string
  name: string
  category?: string
}

export interface FundGraphEdge {
  source: string
  target: string
  value: number
  label: string
}

export interface FundVizData {
  fund_events: FundTimelineEvent[]
  call_events: FundTimelineEvent[]
  anomaly_events: FundTimelineEvent[]
  graph_nodes: FundGraphNode[]
  graph_edges: FundGraphEdge[]
}

export interface TripPoint {
  person_id: string
  lat: number
  lng: number
  ts: string
  weight?: number
}

export interface TripCoOccurrence {
  person_a: string
  person_b: string
  lat: number
  lng: number
  ts: string
  distance_m?: number | null
}

export interface HeatmapCell {
  lng: number
  lat: number
  value: number
}

export interface TripBounds {
  min_lng: number
  max_lng: number
  min_lat: number
  max_lat: number
}

export interface TripVizData {
  points: TripPoint[]
  co_occurrence: TripCoOccurrence[]
  heatmap_cells: HeatmapCell[]
  bounds: TripBounds
}

export function getAnalysisFundViz(edgeLimit = 500): Promise<FundVizData> {
  return http.get('/analysis/fund', { params: { edge_limit: edgeLimit } }) as Promise<FundVizData>
}

export function getAnalysisTripViz(): Promise<TripVizData> {
  return http.get('/analysis/trip') as Promise<TripVizData>
}

File: backend/app/schemas/analysis_viz.py
"""多维可视化 API 契约：资金时间线+流向图、轨迹热力。"""

from typing import Any

from pydantic import BaseModel, Field


class FundTimelineEvent(BaseModel):
    ts: str = Field(description="ISO8601 时间")
    kind: str = Field(description="fund | call | anomaly")
    label: str = ""
    amount: float | None = None
    from_party: str | None = None
    to_party: str | None = None
    meta: dict[str, Any] = Field(default_factory=dict)


class FundGraphNode(BaseModel):
    id: str
    name: str
    category: str = "account"


class FundGraphEdge(BaseModel):
    source: str
    target: str
    value: float = Field(description="金额，用于边宽")
    label: str = ""


class FundVizData(BaseModel):
    """GET /analysis/fund"""

    fund_events: list[FundTimelineEvent]
    call_events: list[FundTimelineEvent]
    anomaly_events: list[FundTimelineEvent]
    graph_nodes: list[FundGraphNode]
    graph_edges: list[FundGraphEdge]


class TripPoint(BaseModel):
    person_id: str
    lat: float
    lng: float
    ts: str
    weight: float = 1.0


class TripCoOccurrence(BaseModel):
    person_a: str
    person_b: str
    lat: float
    lng: float
    ts: str
    distance_m: float | None = None


class HeatmapCell(BaseModel):
    lng: float
    lat: float
    value: float = Field(description="聚合强度")


class TripBounds(BaseModel):
    min_lng: float
    max_lng: float
    min_lat: float
    max_lat: float


class TripVizData(BaseModel):
    """GET /analysis/trip"""

    points: list[TripPoint]
    co_occurrence: list[TripCoOccurrence]
    heatmap_cells: list[HeatmapCell]
    bounds: TripBounds
